### PR TITLE
Activation benchmarks

### PR DESCRIPTION
**Pull request by Priya2698**
- `Gelu` forward and backward: Ports https://github.com/NVIDIA/Fuser/blob/main/benchmark/gelu_backward.cpp
- `Scale-bias-relu` forward and backward: Ports SBR benchmark at https://github.com/NVIDIA/Fuser/blob/main/benchmark/scale_bias_relu.cpp.

### CODE REVIEW

**Comment by Priya2698** [comment]
Merge after #1152

**Comment by naoyam** [comment]
@jacobhinkle Please review

File: python_benchmarks/test_gelu.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    fd: FusionDefinition,
+    dtype: DataType,
+) -> None:
+    input = fd.define_tensor(
+        shape=[-1, -1, -1], contiguity=[True, True, True], dtype=dtype, is_cpu=False
+    )
+    grad = fd.define_tensor(
+        shape=[-1, -1, -1], contiguity=[True, True, True], dtype=dtype, is_cpu=False
+    )
+    bias = fd.define_tensor(shape=[-1], contiguity=[True], dtype=dtype, is_cpu=False)
```
Comment (jacobhinkle) [inline python_benchmarks/test_gelu.py:51]: Do the input/gradient need to be 3D? If not, doing it with 2D inputs instead would cut down the time required to run the benchmarks I think. I know 2D vs 3D shapes are generated differently though so of it's important to do 3D then ignore this.

File: python_benchmarks/test_scale_bias_relu.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    S10 = fd.define_scalar(0.00000, dtype=DataType.Double)
+    T11 = fd.ops.where(T1, T2, S10)
+    T15 = fd.ops.mul(T11, T8)
+
+    if dtype in PROMOTE_DTYPES:
+        T15 = fd.ops.cast(T15, dtype=dtype)
+    fd.add_output(T15)
+
+
+@pytest.mark.parametrize("size", generate_input_sizes(dims=4))
```
Comment (jacobhinkle) [inline python_benchmarks/test_scale_bias_relu.py:79]: Same question as above about whether 4D test is needed vs 2D.

File: python_benchmarks/test_scale_bias_relu.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    with FusionDefinition() as fd:
+        sbr_fwd_fusion(fd, torch_dtype_to_nvfuser_dtype(dtype))
+    if not disable_validation:
+        eager_output = torch.nn.functional.relu(inputs * scale + bias)
+        bool_mask = torch.gt(inputs * scale + bias, 0.0)
+        fd.validate([bias, scale, inputs], [bool_mask, eager_output])
+
+    if not disable_benchmarking:
+        run_benchmark(benchmark, fd.execute, [bias, scale, inputs])
```
Comment (jacobhinkle) [inline python_benchmarks/test_scale_bias_relu.py:100]: This benchmark is failing to compile for me (the bwd benchmark works). Are you seeing this error?
> RuntimeError: kernel->inputs().size() <= args.size() INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/executor_utils.cpp":720, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. KernelArgumentHolder contains less argument than kernel's input.

**Review by jacobhinkle — COMMENTED** [review]
Looks good, but I hit an error on the SBR fwd benchmark. It looks like it might be a separate bug. I can take another look tomorrow morning.

File: python_benchmarks/test_scale_bias_relu.py (hunk 3 above)
Comment (jacobhinkle) [reply python_benchmarks/test_scale_bias_relu.py:100]: Pretty strange! The repro it prints only shows it receiving two inputs even though you've clearly passed all three:
```c++
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id1(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.Float, is_cpu=False)
    T1 = fd.define_tensor(shape=[-1], contiguity=[True], dtype=DataType.Float, is_cpu=False)
    T2 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, True, True], dtype=DataType.Float, is_cpu=False)
    V3 = fd.ops.shape(T2)
    T4 = fd.ops.broadcast_in_dim(T1, shape=V3, broadcast_dims=[3])
    T5 = fd.ops.mul(T2, T4)
    T6 = fd.ops.broadcast_in_dim(T0, shape=V3, broadcast_dims=[3])
    T7 = fd.ops.add(T5, T6)
    S8 = fd.define_scalar(0.00000, dtype=DataType.Double)
    T9 = fd.ops.gt(T7, S8)
    T10 = fd.ops.where(T9, T7, S8)
    fd.add_output(T9)
    fd.add_output(T10)

with FusionDefinition() as fd:
    nvfuser_fusion_id1(fd)

inputs = [
    torch.randn((4194304,), dtype=torch.float32, device='cuda:0').as_strided((128, 64, 32, 16), (32768, 512, 16, 1)),
    torch.randn((16,), dtype=torch.float32, device='cuda:0').as_strided((16,), (1,)),
]
fd.execute(inputs)
```

Comment (Priya2698) [reply python_benchmarks/test_scale_bias_relu.py:100]: This is weird, it is running on local. Let me try on a new docker container.

File: python_benchmarks/test_gelu.py (hunk 1 above)
Comment (Priya2698) [reply python_benchmarks/test_gelu.py:51]: I kept the shapes same as CPP benchmarks for parity.

File: python_benchmarks/test_scale_bias_relu.py (hunk 3 above)
Comment (Priya2698) [reply python_benchmarks/test_scale_bias_relu.py:100]: I am not able to reproduce this error.

Comment (jacobhinkle) [reply python_benchmarks/test_scale_bias_relu.py:100]: I must have screwed up my container. I just pulled into a new container and it works great. Sorry for the false alarm!

**Review by jacobhinkle — APPROVED** [review]

**Pull request merged**